src/commands/admin/admin-subcommands/blacklist.ts

```diff
@@ -0,0 +1,216 @@
+            .addComponents(yes, no);
+
+        const prompt: Message = await ctx.followUp({
+            embeds: [embed],
+            components: [row]
+        });
+
+        // Check if the user pressing the button is the same user
+        const check: CollectorFilter<[ButtonInteraction]> = (m: ButtonInteraction): boolean => m.user.id === ctx.user.id;
+
+        // Disable the buttons to signify that the interaction has ended
+        row.components.forEach(c => c.setDisabled(true));
+
+        try {
+            const confirmation: MessageComponentInteraction = await prompt.awaitMessageComponent({ filter: check, time: 10_000 });
+
+            // End the interaction as soon as a button is pressed
+            await confirmation.update({
+                components: [row]
+            });
+
+            // If the user selected no, abort blacklisting
+            if (confirmation.customId === "no") {
+                await ctx.followUp("Blacklisting denied, aborting opertation");
+                return;
+            }
+        } catch (e) {
+            // End the interaction upon timeout
+            await prompt.edit({
+                components: [row]
+            });
+            // If the interaction timed out, abort as well
+            await ctx.followUp("Interaction timed out, aborting operation");
+            return;
+        }
+
+        // Proceed if the user pressed yes
+        await ctx.followUp(`Commencing blacklisting of \`${user.id}\` (<@${user.id}>)...`);
+
+        // All the servers that the bot is in
+        const servers: Collection<string, Guild> = ctx.client.guilds.cache;
+
+        // The report on whether it was successful or partially successful
+        let report: string = `The user with ID \`${uid}\` (<@${user.id}>) has been blacklisted`;
+        const fails: Guild[] = [];
+
+        for await (const s of servers) {
+            const server: Guild = s[1];
+            try {
+                // Try to ban the user from the server
+                await server.bans.create(
+                    user,
+                    {
+                        reason: reason,
+                        deleteMessageSeconds: BAN_DELETE_SECONDS
+                    }
+                );
+                
+                // Confirm ban if successful
+                await ctx.channel.send(`Banned **${user.username}** from **${server.name}**`);
+            } catch (e) {
+                // Add server to the list of failed bans
+                fails.push(server);
+
+                // Log information if possible
+                if (e instanceof DiscordAPIError && (<DiscordAPIError> e).code === MISSING_PERMISSIONS) {
+                    await ctx.channel.send(`Missing permissions in **${server.name}**, contact <@${server.ownerId}>`);
+                } else {
+                    await ctx.channel.send(`Something went wrong in **${server.name}**, contact <@${server.ownerId}>`);
+                }
+            }
+        }
```

Comment:
This is the bulk of the blacklisting process. Sushi Bot either bans the user from every server it is currently in or reports that the ban didn't go through